src/ShareRoutes.jsx

```diff
@@ -32,6 +33,79 @@ function Forward({appPrefix}) {
 }
 
 
+/**
+ * Check if input is a url
+ * @param {input} input
+ * @return {boolean} return true if url is found
+ */
+export function isURL(input) {
+  assertDefined(input)
+  return input.includes('https://') || input.includes('www')
+}
+
+
+/**
+ * Check if input is a valid url
+ * @param {Object} input
+ * @return {boolean} return true if url is found
+ */
+export function isValidModelURL(input) {
+  assertDefined(input)
+  try {
+    const url = new URL(input)
+    const parsed = new URL(url)
+    if (parsed.hostname !== 'github.com' &&
+        parsed.hostname !== 'www.github.com' &&
+        parsed.hostname !== 'raw.githubusercontent.com') {
+      return false
+    }
+    return true
+  } catch (e) {
+    return false
+  }
+}
+
+
+/**
+ * ParseIfcURL parses Github repository URLs or raw content URLs
+ * @param {string} url
+ * @return {string} Structured path to the model repositore
+ */
+export const parseIfcURL = (url) => {
+  assertDefined(url)
+  const parsed = new URL(url)
+  const parts = parsed.pathname.split('/')
```

Comment:
So I don't think this approach is very safe.  Remember that paths can be deep.. after the repo/branch etc. the file can be simple like "file.ifc" or "a/b/c/d/e/f/g/h/i/file.ifc"

Below then, you do path: parts[5], which I think currently is a bug.  You could join the rest together.

I also proposed a regex with named capture groups in chat that looks significantly simpler than this.

Reply:
right - that makes sense